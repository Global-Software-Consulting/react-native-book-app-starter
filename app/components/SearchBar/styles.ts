import { StyleSheet } from 'react-native';
import { useTheme } from 'react-native-paper';
import {
    heightPercentageToDP as hp,
    widthPercentageToDP as wp,
} from 'react-native-responsive-screen';
import { getPercentageWidth, getPercentageHeight } from 'utils/dimentionUtil';

export const useStyles = () => {
    const theme = useTheme();
    const width = getPercentageWidth();

    const styles = StyleSheet.create({
        container: {
            flex: 1,
            paddingVertical: 10,
            paddingHorizontal: 30,
            backgroundColor: theme.colors.background,
            marginBottom: hp('10%'),
        },
        favoriteView: {
            flex: 1,
            alignItems: 'center',
            justifyContent: 'center',
        },
        bookmark: {
            margin: 2,
        },
        imageError: {
            height: hp('50%'),
            width: wp('50%'),
            tintColor: theme.colors.text,
        },
        mainViewSetting: { height: hp('100%') },
        name: {
            fontSize: hp('3%'),
            fontWeight: 'bold',
            fontFamily: 'Avenir-Medium',
            alignItems: 'center',
            color: theme.colors.text,
        },
        nameShimmer: {
            width: wp('25%'),
            height: hp('6%'),
            backgroundColor: '#E7E5E7',
            marginBottom: 2,
        },
        tagLine: {
            fontSize: hp('2%'),
            fontFamily: 'Avenir-Medium',
            alignItems: 'center',
            color: theme.colors.text,
        },
        tagLineShimmer: {
            width: wp('40%'),
            height: hp('4%'),
            backgroundColor: '#E7E5E7',
        },
        listCaption: {
            marginTop: 15,
            fontSize: hp('2%'),
            fontFamily: 'Avenir-Medium',
            alignItems: 'center',
            fontWeight: 'bold',
            color: theme.colors.text,
        },
        listCaptionShimmer: {
            width: wp('20%'),
            height: hp('3%'),
            backgroundColor: '#E7E5E7',
        },
        button: {
            backgroundColor: '#61e3a5',
            padding: 10,
            borderRadius: 10,
            margin: 10,
        },

        horizontalRuler: {
            borderBottomColor: '#DCDCDC',
            borderBottomWidth: 1,
            marginTop: 3,
            color: theme.colors.primary,
        },
        flatList: {
            flexGrow: 1,
            marginTop: 10,
            justifyContent: 'center',
        },
        flatListLast: {
            flexGrow: 1,
            marginTop: 10,
            marginBottom: hp('3%'),
            justifyContent: 'center',
        },
        searchView: {
            marginTop: 5,
            marginBottom: 5,
            flexDirection: 'row',
            borderWidth: 1,
            borderRadius: 20,
            alignSelf: 'center',
            width: width('90%'),
            borderColor: 'grey',
            justifyContent: 'space-between',
            alignItems: 'center',
            padding: 5,
        },
        searchViewInput: {
            margin: 2,
            marginLeft: 5,
            color: theme.colors.text,
            width: wp('72%'),
        },
        searchViewShimmer: {
            marginBottom: 20,
            flexDirection: 'row',
            borderRadius: 30,
            justifyContent: 'space-between',
            alignItems: 'center',
            height: hp('5%'),
            backgroundColor: '#E7E5E7',
        },
        searchViewChildren: {
            margin: 2,
            marginLeft: 5,
            color: theme.colors.text,
        },
        books: {
            height: hp('2%'),
            width: wp('1%'),
            borderRadius: 4,
        },

        bookTrending: {
            height: hp('25%'),
            width: wp('37%'),
            borderRadius: 4,
        },

        bookGeneral: {
            height: hp('20%'),
            width: wp('30%'),

            borderRadius: 4,
        },
        bookView: {
            backgroundColor: 'transparent',
            paddingRight: 20,
        },
    });
    return styles;
};
