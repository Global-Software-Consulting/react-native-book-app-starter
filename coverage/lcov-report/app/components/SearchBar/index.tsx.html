
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for app/components/SearchBar/index.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> / <a href="index.html">app/components/SearchBar</a> index.tsx</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">91.66% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>11/12</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">80% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>4/5</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">91.66% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>11/12</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line high'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { useState } from 'react'
import { useTranslation } from 'react-i18next'
import { TextInput, TouchableOpacity, View } from 'react-native'
import { useTheme } from 'react-native-paper'
import Icon from 'react-native-vector-icons/FontAwesome'
import { useDispatch } from 'react-redux'
import * as appActions from 'store/actions/appActions'
import { useStyles } from './styles'
&nbsp;
const SearchBar: React.FC = () =&gt; {
    const [searchText, setSearchText] = useState&lt;string&gt;('')
    const theme = useTheme()
    const styles = useStyles()
    const dispatch = useDispatch()
    const { t } = useTranslation()
&nbsp;
    const fetchBookDetails = async () =&gt; {
        dispatch(appActions.getBookRequest(searchText))
    }
&nbsp;
    return (
        &lt;View style={styles.searchView}&gt;
            &lt;TextInput
                testID={'searchInput'}
                underlineColorAndroid="transparent"
                placeholder={t('Search Here')}
                placeholderTextColor={theme?.colors?.text}
                onChangeText={(text) =&gt; setSearchText(text)}
                style={styles.searchViewInput}
                onEndEditing={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >fetchBookDetails()}</span>
            /&gt;
            &lt;TouchableOpacity testID={'searchBookies'} onPress={() =&gt; fetchBookDetails()}&gt;
                &lt;Icon name="search" size={20} style={styles.searchViewChildren} /&gt;
            &lt;/TouchableOpacity&gt;
        &lt;/View&gt;
    )
}
export default SearchBar
&nbsp;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
<<<<<<< HEAD
                at Thu Nov 11 2021 14:22:06 GMT+0500 (Pakistan Standard Time)
=======
                at Wed Nov 10 2021 16:06:07 GMT+0500 (Pakistan Standard Time)
>>>>>>> 44ebcc3db372a8f9d63d4872f4fe55409e5c797f
            </div>
        </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    