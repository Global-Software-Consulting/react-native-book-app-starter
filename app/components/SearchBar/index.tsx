import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { TextInput, TouchableOpacity, View } from 'react-native';
import { useTheme } from 'react-native-paper';
import Icon from 'react-native-vector-icons/FontAwesome';
import { useDispatch } from 'react-redux';
import * as appActions from 'store/actions/appActions';
import { useStyles } from './styles';

const SearchBar: React.FC = () => {
    const [searchText, setSearchText] = useState<string>('');
    const theme = useTheme();
    const styles = useStyles();
    const dispatch = useDispatch();
    const { t } = useTranslation();

    const fetchBookDetails = async () => {
        dispatch(appActions.getBookRequest(searchText));
    };

    return (
        <View style={styles.searchView}>
            <TextInput
                testID={'searchInput'}
                underlineColorAndroid="transparent"
                placeholder={t('Search Here')}
                placeholderTextColor={theme.colors.text}
                onChangeText={(text) => setSearchText(text)}
                style={styles.searchViewInput}
                onEndEditing={() => fetchBookDetails()}
            />
            <TouchableOpacity testID={'searchBookies'} onPress={() => fetchBookDetails()}>
                <Icon name="search" size={20} style={styles.searchViewChildren} />
            </TouchableOpacity>
        </View>
    );
};
export default SearchBar;
