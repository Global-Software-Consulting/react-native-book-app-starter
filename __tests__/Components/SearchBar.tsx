import React from 'react'
import { fireEvent, render } from '@testing-library/react-native'
import SearchBar from './../../app/components/SearchBar/index'

describe('Component testing', () => {
    test('Snapshot', () => {
        const tree = render(<SearchBar />)
        expect(tree).toMatchSnapshot()
    })
    test('search action', () => {
        const { getByTestId } = render(<SearchBar />)
        // fireEvent.press(getByTestId('search'));
        fireEvent.press(getByTestId('searchBookies'))
    })
    test('search input change', () => {
        const { getByTestId } = render(<SearchBar />)
        // fireEvent.press(getByTestId('search'));
        fireEvent.changeText(getByTestId('searchInput'))
    })
    test('search input change', () => {
        const { getByTestId } = render(<SearchBar />)
        // fireEvent.press(getByTestId('search'));
    })
})
