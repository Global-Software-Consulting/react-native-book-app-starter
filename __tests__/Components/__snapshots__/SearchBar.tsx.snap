// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component testing Snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "borderColor": "grey",
      "borderRadius": 20,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 5,
      "marginTop": 5,
      "padding": 5,
      "width": 675,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    onEndEditing={[Function]}
    placeholderTextColor="#000000"
    rejectResponderTermination={true}
    style={
      Object {
        "color": "#000000",
        "margin": 2,
        "marginLeft": 5,
        "width": undefined,
      }
    }
    testID="searchInput"
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="searchBookies"
  >
    <mockedDefaultExport
      name="search"
      size={20}
      style={
        Object {
          "color": "#000000",
          "margin": 2,
          "marginLeft": 5,
        }
      }
    />
  </View>
</View>
`;
